src/main/java/com/icthh/xm/ms/entity/domain/XmEntity.java

```diff
@@ -264,6 +266,12 @@
     @Getter @Setter
     private Set<UniqueField> uniqueFields = new HashSet<>();
 
+    @OneToOne(mappedBy = "eventDataRef", fetch = LAZY)
+    @JsonIgnore
+    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+    @Getter @Setter
+    private Event event;
```

Comment:
@abonert do we really need this field from XM entity side here?
As we discussed this reference was added to handle XMEntity deletion (unlink Event from the deleted entity)
Alternative as proposed by @abonert is to use EventRepository to find the Event and then unlink within it.

@sergeysenja1992 @major13ua what is your vision?

Reply:
i agree with @amedvedchuk , we don't need link to event in XmEntity

Reply:
@abonert please try to use the second option then.

Reply:
pushed. Please [review](https://github.com/xm-online/xm-ms-entity/pull/217/commits/c473f90d053b7320475395d080e2ec95caeda834)